### ISSUE TITLE
chore: Update go version to 1.18 at least

### ISSUE THREAD

**Issue by dmvolod** [kubewarden/k8s-objects-generator#16]
### Is your feature request related to a problem?

As most of the `Kubewarden` dependent projects already on golang 1.18 and 1.19, we could probably bootstrap golang version for generator and generating objects to the 1.18 (1.19?)

### Solution you'd like

Update golang to the 1.18(1.19?) and use `github.com/deckarep/golang-set/v2` (refactoring with generics need to be implemented)

**Comment by flavio** [comment]
Sure, that looks good to me. Let me know if you want to work on that

**Comment by dmvolod** [comment]
Yes, working on it, but seems to we don't need to upgrade golang version for generating objects for now
